edweiss-app/app/(app)/calendar/index.tsx

```diff
@@ -6,85 +6,119 @@ import { useAuth } from '@/contexts/auth';
 import { useDynamicDocs } from '@/hooks/firebase/firestore';
 import { Course } from '@/model/school/courses';
 import React, { useEffect, useRef, useState } from 'react';
-import { ActivityIndicator, Animated, Dimensions, FlatList, View } from 'react-native';
+import { ActivityIndicator, Animated, Dimensions, FlatList, ScaledSize, View } from 'react-native';
+export type CourseWithId = Course & { id: string };
 
-const InfinitePaginatedCounterScreen = () => {
-  const auth = useAuth();
-  const [loading, setLoading] = useState(true);
-  const [pages, setPages] = useState(Array.from({ length: 10 }, (_, index) => index - 3)); // From -7 to +7
-  const scrollRef = useRef<FlatList>(null);
-  const [dimensions, setDimensions] = useState(Dimensions.get('window'));
+const useWindowDimensions = () => {
+  const [dimensions, setDimensions] = useState(() => Dimensions.get('window'));
 
-  // Global listener for dimension changes
   useEffect(() => {
-    const handleResize = ({ window }: { window: { width: number; height: number; scale: number; fontScale: number } }) => setDimensions(window);
+    const handleResize = ({ window }: { window: ScaledSize }) =>
+      setDimensions(window);
+
     const subscription = Dimensions.addEventListener('change', handleResize);
     return () => subscription?.remove();
   }, []);
 
-  // Loading courses and tasks
+  return dimensions;
+};
+
+const useCourseData = (authUserId: string) => {
+  const courses: CourseWithId[] = useDynamicDocs(CollectionOf('courses'))?.map(doc => ({ ...doc.data as unknown as Course, id: doc.id })) ?? [];
+
+
   const myCourses = useDynamicDocs(
-    CollectionOf(`users/${auth.authUser?.uid ?? 'default-uid'}/courses`)
-  )?.map(doc => ({ id: doc.id, data: doc.data })) ?? [];
+    CollectionOf(`users/${authUserId}/courses`)
+  )?.map((doc) => ({ id: doc.id, data: doc.data })) ?? [];
 
-  const courses = useDynamicDocs(CollectionOf('courses'))?.map(doc => ({
-    id: doc.id,
-    data: doc.data as unknown as Course
-  })) ?? [];
+  const myCourseIds = myCourses.map((course) => course.id);
+  const filteredCourses = courses.filter((course) => myCourseIds.includes(course.id));
+
+  return { courses, filteredCourses };
+};
 
+const useAssignmentsAndTodos = (authUserId: string, courses: Course[]) => {
+  const [loading, setLoading] = useState(true);
   const [assignmentsByCourse, setAssignmentsByCourse] = useState<{ [key: string]: any[] }>({});
   const [todoByCourse, setTodoByCourse] = useState<{ [key: string]: any[] }>({});
 
-  // Fetch assignments and todos
   useEffect(() => {
     const fetchData = async () => {
-      const assignmentsData: { [key: string]: any[] } = {};
-      const todoData: { [key: string]: any[] } = {};
+      if ((courses as unknown as CourseWithId[]).length === 0) return;
```

Comment:
Why is this specific "if" using 3 equal signs? Isn't === useful for strict equality? I don't know if that would change much from the == since you are comparing 2 numbers. But I could be wrong since I do not particularly understand === , if so do not hesitate to correct my statement.

Reply:
=== compares both the value and the type, ensuring that no type conversion happens.